internal/strace/strace.go

```diff
@@ -132,22 +132,16 @@ func parseCmdAndEnv(cmdAndEnv string) ([]string, []string, error) {
 	return cmd, env, nil
 }
 
-func (r *Result) recordFileAccess(file string, read, write, delete bool) {
+func (r *Result) recordFileAccess(file string, read, write, delete bool, bytesWritten int64) {
 	if _, exists := r.files[file]; !exists {
 		r.files[file] = &FileInfo{Path: file}
 	}
 	r.files[file].Read = r.files[file].Read || read
 	r.files[file].Write = r.files[file].Write || write
 	r.files[file].Delete = r.files[file].Delete || delete
-}
 
-func (r *Result) recordFileWrite(file string, bytesWritten int64) {
-	if bytesWritten > 0 {
-		if _, exists := r.files[file]; !exists {
-			r.files[file] = &FileInfo{Path: file, WriteInfo: []WriteContentInfo{}}
-		}
+	if write && bytesWritten > 0 {
```

Comment:
Is it worth logging a warning if` `write == false && bytesWritten > 0`? 

Moving forward, perhaps `write` is now redundant if it is equivalent to `bytesWritten == 0`

Reply:
I think my original comment was not to just use `recordFileAccess` for everything, but just call `recordFileAccess` from `recordFileWrite` to initialize the `r.files[file]` if needd. 

It's best to keep them separate, and it would avoid the issue Max descibed.

Reply:
Gotcha, that makes sense. I'll put up a modification for this.